src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py

```diff
@@ -202,7 +202,77 @@ def test_aks_bad_inputs(self, resource_group, resource_group_location, sp_name,
             self.cmd(create_cmd)
         self.assertIn('--max-pods', str(err.exception))
+    
+    @ResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='eastus')
+    @RoleBasedServicePrincipalPreparer()
+    def test_aks_create_default_service_without_skip_role_assignment(self, resource_group, resource_group_location, sp_name, sp_password):
+        aks_name = self.create_random_name('cliakstest', 16)
+        self.kwargs.update({
+            'resource_group': resource_group,
+            'name': aks_name,
+            'location': resource_group_location,
+            'service_principal': sp_name,
+            'client_secret': sp_password,
+            'vnet_subnet_id': self.generate_vnet_subnet_id(resource_group),
+            'resource_type': 'Microsoft.ContainerService/ManagedClusters'
+        })
+        # create cluster without skip_role_assignment
+        create_cmd = 'aks create --resource-group={resource_group} --name={name} --location={location} ' \
+                     '--node-count=1 --service-principal={service_principal} ' \
+                     '--client-secret={client_secret} --vnet-subnet-id={vnet_subnet_id}'
+        self.cmd(create_cmd, checks=[
+            self.check('agentPoolProfiles[0].vnetSubnetId', '{vnet_subnet_id}'),
+            self.check('provisioningState', 'Succeeded')
+        ])
+
+        check_role_assignment_cmd = 'role assignment list --scope={vnet_subnet_id}'
+        self.cmd(check_role_assignment_cmd, checks=[
+            self.check('[0].scope', '{vnet_subnet_id}')
+        ])
+
+        # create cluster with same role assignment
+        aks_name = self.create_random_name('cliakstest', 16)
+        self.kwargs.update({
+            'name': aks_name,
+        })
+
+        create_cmd = 'aks create --resource-group={resource_group} --name={name} --location={location} ' \
+                     '--node-count=1 --service-principal={service_principal} ' \
+                     '--client-secret={client_secret} --vnet-subnet-id={vnet_subnet_id}'
+        self.cmd(create_cmd, checks=[
+            self.check('agentPoolProfiles[0].vnetSubnetId', '{vnet_subnet_id}'),
+            self.check('provisioningState', 'Succeeded')
+        ])
 
+    
+    @ResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='eastus')
+    @RoleBasedServicePrincipalPreparer()
+    def test_aks_create_default_service_with_skip_role_assignment(self, resource_group, resource_group_location, sp_name, sp_password):
+        aks_name = self.create_random_name('cliakstest', 16)
+        self.kwargs.update({
+            'resource_group': resource_group,
+            'name': aks_name,
+            'location': resource_group_location,
+            'service_principal': sp_name,
+            'client_secret': sp_password,
+            'vnet_subnet_id': self.generate_vnet_subnet_id(resource_group),
+            'resource_type': 'Microsoft.ContainerService/ManagedClusters'
```

Comment:
You don't use the `resource_type` kwarg, you can remove it.

Reply:
why not need this? is this will be the default one?

Reply:
never mind, you are right.